### ISSUE TITLE
Define what to report in compatibles for non arm devices

### ISSUE THREAD

**Issue by helen-fornazier** [kernelci/kernelci-project#478]
We need to think about what to report on compatible for x86, otherwise we get hardware unknown  [https://staging.dashboard.kernelci.org:9000/tree/6fb2fa9805c501d9ade047fc511961f3273cdcb5/test[…]3A%226fb2fa9805c501d9ade047fc511961f3273cdcb5%22%7D](https://staging.dashboard.kernelci.org:9000/tree/6fb2fa9805c501d9ade047fc511961f3273cdcb5/test/maestro%3A6720808faaccbda5366098da?tableFilter=%7B%22buildsTable%22%3A%22all%22%2C%22bootsTable%22%3A%22failed%22%2C%22testsTable%22%3A%22all%22%7D&origin=maestro&currentTreeDetailsTab=treeDetails.boots&diffFilter=%7B%7D&treeInfo=%7B%22gitBranch%22%3A%22master%22%2C%22gitUrl%22%3A%22https%3A%2F%2Fgit.kernel.org%2Fpub%2Fscm%2Flinux%2Fkernel%2Fgit%2Fnext%2Flinux-next.git%22%2C%22treeName%22%3A%22next%22%2C%22commitName%22%3A%22next-20241029%22%2C%22headCommitHash%22%3A%226fb2fa9805c501d9ade047fc511961f3273cdcb5%22%7D)
and it is hard to understand from the main page which are the boards that are failing to boot
example: [https://staging.dashboard.kernelci.org:9000/tree/6fb2fa9805c501d9ade047fc511961f3273cdcb5?tabl[…]3A%226fb2fa9805c501d9ade047fc511961f3273cdcb5%22%7D](https://staging.dashboard.kernelci.org:9000/tree/6fb2fa9805c501d9ade047fc511961f3273cdcb5?tableFilter=%7B%22buildsTable%22%3A%22all%22%2C%22bootsTable%22%3A%22failed%22%2C%22testsTable%22%3A%22all%22%7D&origin=maestro&currentTreeDetailsTab=treeDetails.boots&diffFilter=%7B%7D&treeInfo=%7B%22gitBranch%22%3A%22master%22%2C%22gitUrl%22%3A%22https%3A%2F%2Fgit.kernel.org%2Fpub%2Fscm%2Flinux%2Fkernel%2Fgit%2Fnext%2Flinux-next.git%22%2C%22treeName%22%3A%22next%22%2C%22commitName%22%3A%22next-20241029%22%2C%22headCommitHash%22%3A%226fb2fa9805c501d9ade047fc511961f3273cdcb5%22%7D)

cc @nuclearcat @spbnick @JenySadadia @padovan

**Comment by spbnick** [comment]
Yes, and it cannot be the `compatible` field. The best I found so far is getting the ACPI info. Check out what you can see under `/sys/class/dmi/id/` - I think it's a good candidate.

**Comment by helen-fornazier** [comment]
```
ls -1 /sys/class/dmi/id/                                                                                   (base) 
bios_date
bios_release
bios_vendor
bios_version
board_asset_tag
board_name
board_serial
board_vendor
board_version
chassis_asset_tag
chassis_serial
chassis_type
chassis_vendor
chassis_version
modalias
power/
product_family
product_name
product_serial
product_sku
product_uuid
product_version
subsystem@
sys_vendor
uevent
```

Maybe we could have a composition of a few of those?

**Comment by spbnick** [comment]
Yeah, we can pick a bunch of files we like and convert them into correspondingly-named fields under `tests[<id>].environment.dmi`. We can start with these, for example:

* `sys_vendor`
* `board_vendor`
* `board_name`
* `product_family`
* `product_name`

**Comment by spbnick** [comment]
Ideally, retrieval of this info should be done on the machine before test execution, but prefilling Maestro HW database with these would also work, similar to how you set `compatible`.

**Comment by spbnick** [comment]
@helen-fornazier, @nuclearcat, the sooner we agree on a plan, the sooner I can start writing the schema and the proposal, and working on the implementation. It's obvious, of course, just wanted to let you know, that I'm waiting on you :grin:
